bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,239 @@
+                        tob(zonedDateTime.getSecond()), tob(zonedDateTime.getDayOfWeek().getValue()))
+                                .execute(floureonDevice);
+            } catch (Exception e) {
+                logger.error("Error while setting time of {} to {}", thing.getUID(), command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    @Override
+    protected void refreshData() {
+        try {
+            AdvancedStatusInfo advancedStatusInfo = floureonDevice.getAdvancedStatus();
+            if (advancedStatusInfo == null) {
+                logger.warn("Device {} did not return any data. Trying to reauthenticate...", thing.getUID());
+                authenticate();
+                advancedStatusInfo = floureonDevice.getAdvancedStatus();
+            }
+            if (advancedStatusInfo == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Device not responding.");
+                return;
+            }
+            logger.debug("Retrieved data from device {}: {}", thing.getUID(), advancedStatusInfo);
+            logger.debug("Updating channel {} with value {}", ROOM_TEMPERATURE,
+                    new DecimalType(advancedStatusInfo.getRoomTemp()));
+            logger.debug("Mode {}", StringType.valueOf(advancedStatusInfo.getAutoMode() ? "auto" : "manual"));
+            updateState(ROOM_TEMPERATURE, new DecimalType(advancedStatusInfo.getRoomTemp()));
+            logger.debug("Updating channel {} with value {}", ROOM_TEMPERATURE_EXTERNAL_SENSOR,
+                    new DecimalType(advancedStatusInfo.getExternalTemp()));
+            updateState(ROOM_TEMPERATURE_EXTERNAL_SENSOR, new DecimalType(advancedStatusInfo.getExternalTemp()));
+            updateState(SETPOINT, new DecimalType(advancedStatusInfo.getThermostatTemp()));
+            updateState(POWER, OnOffType.from(advancedStatusInfo.getPower()));
+            updateState(MODE, StringType.valueOf(advancedStatusInfo.getAutoMode() ? "auto" : "manual"));
+            updateState(SENSOR, StringType.valueOf(advancedStatusInfo.getSensorControl().name()));
+            logger.debug("Updating channel {} with value {}", SENSOR,
+                    new StringType(advancedStatusInfo.getSensorControl().name()));
+            updateState(TEMPERATURE_OFFSET, new DecimalType(advancedStatusInfo.getDif()));
+            updateState(ACTIVE, OnOffType.from(advancedStatusInfo.getActive()));
+            updateState(REMOTE_LOCK, OnOffType.from(advancedStatusInfo.getRemoteLock()));
+            String timestamp = getTimestamp(advancedStatusInfo);
+            logger.debug("Composed timestamp: {}", timestamp);
+            updateState(TIME, DateTimeType.valueOf(timestamp));
+        } catch (Exception e) {
+            logger.error("Error while retrieving data for {}", thing.getUID(), e);
+        }
+    }
+
+    @NotNull
```

Comment:
This annotation has no effect and can be removed.

Reply:
Removed.